src/clients/c/tb_client/context.zig

```diff
@@ -560,6 +562,22 @@ pub fn ContextType(
         }
 
         fn on_submit(implementation: *ContextImplementation, packet: *Packet) void {
+            // Packet is caller-allocated to enable elastic intrusive-link-list-based memory
+            // management. However, some of Packet's fields are essentially private. Initialize
+            // them here to avoid threading default fields through FFI boundary.
+            packet.* = .{
+                .next = null,
+                .user_data = packet.user_data,
+                .operation = packet.operation,
+                .status = .ok,
```

Comment:
Note when refactoring `tb_packet_t`:

`status` is the only field that is both exposed to the client and updated by `tb_client`. We should change this API so that the status is instead returned alongside `results`.